keen/slate_algolia
# Before index hook

## What does this PR do?

It adds a `before_index` option to the extension, which accepts a proc that will be called with each record prior to indexing. This allows users to modify the data format of the records, prior to them being shipped off to Algolias
## How should this be tested?
- Read the tests, make sure they're sane, and run them
- If you're so inclined, you can play around with the `base` fixture to actually run this on a middleman app
## Notes

I did some tab-sizing fixes on this one, so there's a lot of lines changed that didn't actually change functionally. Sry.
